src/libopenrave/kinbody.cpp

```diff
@@ -6031,6 +6031,12 @@ void KinBody::Clone(InterfaceBaseConstPtr preference, int cloningoptions)
 void KinBody::_PostprocessChangedParameters(uint32_t parameters)
 {
     _nUpdateStampId++;
+
+    // If we changed any properties saved in _CreateSaverForGrabber, we need to invalidate the cache
+    if (parameters & (Prop_Links | Prop_JointLimits)) {
```

Comment:
@rschlaikjer 

Thanks for great and interesting idea!

In the `Grabbed` class, for example, the followings `saverOption` is used:
```
    _CreateSaverForGrabbedAndGrabber(_pGrabberSaver,
                                     pGrabber,
                                     KinBody::Save_LinkTransformation|KinBody::Save_LinkEnable|KinBody::Save_JointLimits|KinBody::Save_LinkVelocities, // Need to save link velocities of the grabber since will be used for computing link velocities of the grabbed bodies.
                                     bDisableRestoreOnDestructor);
```

In the state saver, for example of `linkEnable`, the following `KinBody` API is used:
```
    if( _options & Save_LinkEnable ) {
        _vEnabledLinks.resize(_pbody->GetLinks().size());
        for(size_t i = 0; i < _vEnabledLinks.size(); ++i) {
            _vEnabledLinks[i] = _pbody->GetLinks().at(i)->IsEnabled();
        }
    }
```

In the `KinBody::Enable` to mutate such status, it using the folloring `Prop`
```
    if( bchanged ) {
        _PostprocessChangedParameters(Prop_LinkEnable);
    }
```

Therefore, this line should check `Prop_LinkEnable` as well.

Based on the similar observation, several other saver options are not well checked, and several other `Prop` options are not checked as well.

In addition, if I think about `Save_LinkVelocities`, I'm not sure  `SetDOFVelocities` or `SetLinkVelocities` which is the setter function of relevant members are well maintained via `Prop`.

Reply:
Therefore, let's figure out what happens if we use saver options like `KinBody::Save_LinkTransformation|KinBody::Save_LinkEnable|KinBody::Save_JointLimits|KinBody::Save_LinkVelocities`.


- First, could you figure out which `KinBody` API and internal member variables are referred from `KinBodyStateSaver` by them?
- Next, could you figure out which `Prop` options can be used for them?
- Next, could you check if the necessary member variables are all managed by the given `Prop` option? for example, i'm bit questionable that `LinkVelocities` is managed by `Prop`.
- Finally, could you do the enough testing and write the enough test cases to internal repos?

Thanks a lot!